lib/api/dsl/methods.js

```diff
@@ -7,8 +7,202 @@ var
+      _.isString(geoFilter.topLeft) &&
+      _.isString(geoFilter.bottomRight) &&
+      /^[0-9a-z]{4,}$/.test(geoFilter.topLeft) &&
+      /^[0-9a-z]{4,}$/.test(geoFilter.bottomRight)
+    ) {
+      tmp = geohash.decode(geoFilter.topLeft);
+      top = tmp.longitude;
+      left = tmp.latitude;
+
+      tmp = geohash.decode(geoFilter.bottomRight);
+      bottom = tmp.longitude;
+      right = tmp.latitude;
+    }
+
+    if (top && left && bottom && right) {
+      if (!_.isNumber(top)) {
+        top = parseFloat(top);
+      }
+      if (!_.isNumber(left)) {
+        left = parseFloat(left);
+      }
+      if (!_.isNumber(bottom)) {
+        bottom = parseFloat(bottom);
+      }
+      if (!_.isNumber(right)) {
+        right = parseFloat(right);
+      }
+    }
+    else {
+      throw new BadRequestError('Unable to parse coordinates');
+    }
+
+    return {top: top, left: left, bottom: bottom, right: right};
+  },
+
+  /**
+   * Construct a valid usable point
+   *
+   * @param {Object} the given object
+   * @return {Object} the valid usable point object
+   */
+  constructPoint: function(geoFilter) {
+    var lat, lon, tmp;
+
+    // { lat: -74.1, lon: 40.73 }
+    if (geoFilter.lat !== undefined &&
+      geoFilter.lon !== undefined
+    ) {
+      lat = geoFilter.lat;
+      lon = geoFilter.lon;
+    }
+    // { latLon: { lat: 40.73, lon: -74.1 } }
+    else if (geoFilter.latLon &&
+      geoFilter.latLon.lat &&
+      geoFilter.latLon.lon
+    ) {
+      lat = geoFilter.latLon.lat;
+      lon = geoFilter.latLon.lon;
+    }
+    // { latLon: [ -74.1, 40.73 ] }
+    else if (geoFilter.latLon &&
+      _.isArray(geoFilter.latLon) 
+    ) {
+      lat = geoFilter.latLon[0];
+      lon = geoFilter.latLon[1];
+    }
+    // { latLon: "40.73, -74.1" }
+    else if (geoFilter.latLon &&
+      _.isString(geoFilter.latLon) &&
+      /^[-.0-9]+,\s*[-.0-9]+$/.test(geoFilter.latLon)
+    ) {
+      tmp = geoFilter.latLon.match(/^([-.0-9]+),\s*([-.0-9]+)$/);
+      lat = tmp[1];
+      lon = tmp[2];
+    }
+    // { latLon: "dr5r9ydj2y73"}
+    else if (geoFilter.latLon &&
+      _.isString(geoFilter.latLon) &&
+      /^[0-9a-z]{4,}$/.test(geoFilter.latLon) 
+    ) {
+      tmp = geohash.decode(geoFilter.latLon);
+      lat = tmp.latitude;
+      lon = tmp.longitude;
+    }
+
+    if (lat !== undefined && lon !== undefined) {
+      if (!_.isNumber(lat)) {
+        lat = parseFloat(lat);
+      }
+      if (!_.isNumber(lon)) {
+        lon = parseFloat(lon);
+      }
+    }
+    else {
+      throw new BadRequestError('Unable to parse coordinates');
+    }
+    return {lat: lat, lon: lon};
+  },
+
+  /**
+   * Generate a valid usable distance
+   *
+   * @param {String} the given distance
+   * @return {Object} the distance in meters
+   */
+  getDistance: function(distance) {
+    var tmp;
+    if (_.isString(distance)) {
+      tmp = distance.replace(/^([0-9., ]*)([a-z])/, '$1 $2').replace(/ /,'').replace(/,/,'.').replace(/-/,'').toLowerCase();
```

Comment:
1. As-is, the `replace(/-/, '')` looks undeeded to me (not allowed in the first regex).
2. I suppose the first regex should be `/^([0-9., ]+)([a-z]+)/i` to allow uppercase units and let the user enter some multi-char units (i.e. km).